### DISCUSSION TITLE
Custom Data Into Labels

### DISCUSSION THREAD

**Discussion by ydragos** [kernc/backtesting.py#231] [Ideas]
Hi, 
Thank you for a great library.
Can we add custom data into plot labels? I use outside signals data that I bring into the data frame like :Signal1:Long, Signal2:Short and when a few of them are in agreement I fire a trade. I would like to see the values of Signal1,Signal2 etc on the Labels in the graph.

**Comment by kernc** [comment]
I'm not sure what you mean by adding data into labels. Do you mean plotting externally provided signals, such as this?
```py
def init(self):
    # Plot 'Signal1' column from the input dataframe
    self.I(lambda col: self.data.df[col], 'Signal1')
```
See also: https://github.com/kernc/backtesting.py/issues/64.

**Comment by ydragos** [comment]
right, but Signal1 is a string not numeric. I would just want to see it in the label for that day.

**Comment by ydragos** [comment]
so just wanted to see it the label that appears when you hover over a candlestick.

**Reply by kernc** [reply]
You might write an enhancement proposal about it on the issue tracker.

**Comment by kernc** [comment]
The only two current indicator plot types are [line and scatter plot](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I). I guess for categorical variables you could also do something along the lines of:
```py
import pandas as pd

# Convert distinct labels in 'Signal1' to numbers
self.I(lambda col: pd.Categorical(self.data.df[col]).codes, 'Signal1')
```
If you'd like to work on "label" plot type for categoricals, PR welcome.

**Reply by ydragos** [reply]
Thank you so much for your help. Let me see how far I can get with this and if more is needed I will reach out for PR details.
